pkg/nfd-worker/nfd-worker.go

```diff
@@ -713,25 +713,21 @@ func (m *nfdWorker) updateNodeFeatureObject(labels Labels) error {
 
 	features := source.GetAllFeatures()
 
-	// Create owner ref
-	ownerRefs := []metav1.OwnerReference{}
+	// Get pod ower reference
 	podName := os.Getenv("POD_NAME")
-	podUID := os.Getenv("POD_UID")
-	if podName != "" && podUID != "" {
-		isTrue := true
-		ownerRefs = []metav1.OwnerReference{
-			{
-				APIVersion: "v1",
-				Kind:       "Pod",
-				Name:       podName,
-				UID:        types.UID(podUID),
-				Controller: &isTrue,
-			},
-		}
-	} else {
-		klog.InfoS("Cannot set NodeFeature owner reference, POD_NAME and/or POD_UID not specified")
+	client, err := m.getKubeClient()
```

Comment:
should we get owner-ref once on start (eg in main) and pass it to nfdWorker as attribute ?
this will save api call on every update (if such occurs).

Reply:
agree

Reply:
done

Reply:
i see latest version still gets pod owner ref on every call

Reply:
I forgot to push? umm let me check

Reply:
my bad, I forgot to push the last ammend